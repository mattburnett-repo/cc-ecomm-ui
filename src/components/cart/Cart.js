export default function Cart ()  {
    return (
        <div>
            <div role="presentation" aria-label="cart-items">
                <div>cart item 01</div>
                <div>cart item 02</div>
                <div>cart item 03</div>
            </div>
            <div role="presentation" aria-label="cart-total-price">
                <label htmlFor="cart-total-price">Cart Total</label>
                <div id="cart-total-price">cart total goes here</div>
            </div>
            <button aria-label="checkout-button">Checkout</button>
        </div>
    )
}