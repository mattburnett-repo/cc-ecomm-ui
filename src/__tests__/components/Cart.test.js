// 20211217: Wireframe https://wireframe.cc/kXiUKu

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import Cart from '../../components/cart/Cart';

describe('<Cart /> component tests', () => {
    beforeEach(() => render(<Cart />));

    it('should display cart items', () => {
        screen.getByRole('presentation', {name: /cart-items/i});
    });
    it('should display cart total price', () => {
        screen.getByRole('presentation', {name: /cart-total-price/i});
    });
    it('should display a checkout button', () => {
        screen.getByRole('button', {name: /checkout-button/i});
    });
}); // end component

describe('<Cart /> feature tests', () => {
    beforeEach(() => render(<Cart />));

    it('does something else?', () => {
        expect(true).toBeFalsy();
    })

    it('creates a snapshot', () => {
        const tree = renderer.create(<Cart />).toJSON();
        expect(tree).toMatchSnapshot();  
    })
}); // end feature

describe('<Cart /> unit / functional tests', () => {
    beforeEach(() => render(<Cart />));

    it('clicks the checkout button', () => {
        const theVal = screen.getByRole('button', {name: /checkout-button/i});
        fireEvent.click(theVal);
    });
}); // end unit / functional
