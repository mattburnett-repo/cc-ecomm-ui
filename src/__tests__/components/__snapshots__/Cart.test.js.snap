// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> feature tests creates a snapshot 1`] = `
<div>
  <div
    aria-label="cart-items"
    role="presentation"
  >
    <div>
      cart item 01
    </div>
    <div>
      cart item 02
    </div>
    <div>
      cart item 03
    </div>
  </div>
  <div
    aria-label="cart-total-price"
    role="presentation"
  >
    <label
      htmlFor="cart-total-price"
    >
      Cart Total
    </label>
    <div
      id="cart-total-price"
    >
      cart total goes here
    </div>
  </div>
  <button
    aria-label="checkout-button"
  >
    Checkout
  </button>
</div>
`;
